conformance/lang/expressions/logical-expr/logical_or_expr.balt

```diff
@@ -0,0 +1,853 @@
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using typeof expression.
+Labels: logical-expr, typeof-expr
+
+function errorFunction() {
+    boolean _ = true || typeof true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = typeof true || true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = typeof false || typeof true; // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using unary expression.
+Labels: logical-expr, unary-complement, unary-minus, unary-plus
+
+function errorFunction() {
+    boolean _ = true || +2; // @error static type of each expression must be a subtype of boolean
+    boolean _ = -2 || true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = ~2 || !true; // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using multiplicative expression.
+Labels: logical-expr, multiplicative-expr
+
+function errorFunction() {
+    boolean _ = true || 1 * 2; // @error static type of each expression must be a subtype of boolean
+    boolean _ = 1 / 2 || true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = 1 % 2 || 1 * 2; // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using additive expression.
+Labels: logical-expr, additive-expr
+
+function errorFunction() {
+    boolean _ = true || 1 + 2; // @error static type of each expression must be a subtype of boolean
+    boolean _ = 1 - 2 || true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = 1 + 2 || 1 - 2; // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using shift expression.
+Labels: logical-expr, shift-expr
+
+function errorFunction() {
+    boolean _ = true || 1 << 2; // @error static type of each expression must be a subtype of boolean
+    boolean _ = 1 >> 2 || true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = 1 >> 2 || 1 << 2; // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using range expression.
+Labels: logical-expr, range-expr
+
+function errorFunction() {
+    boolean _ = true || (1 ..< 5).iterator().next(); // @error static type of each expression must be a subtype of boolean
```

Comment:
Why do we have to call `.iterator().next()`?

Reply:
[94d5da7](https://github.com/ballerina-platform/ballerina-spec/pull/1113/commits/94d5da7d6a3c741d8d2a8cb2d22ac6910b20f776)